bluesky/protocols.py

```diff
@@ -1,17 +1,54 @@
 try:
-    from typing import Protocol, runtime_checkable
+    from typing_extensions import Literal, Protocol, TypedDict, runtime_checkable
 except ImportError:
-    from typing_extensions import Protocol, runtime_checkable
+    from typing import Literal, Protocol, TypedDict, runtime_checkable
 
-from typing import Dict, Any, Optional, Callable, Generator, List, Union, NoReturn
+from typing import Any, Awaitable, Callable, Dict, Generator, List, Optional, TypeVar, Union
 
+from enum import IntEnum
 
-Configuration = Dict[str, Dict[str, Any]]
+class Severity(IntEnum):
+    #: Value is known, valid, nothing is wrong
+    VALID = 0
+    #: Value is known, valid, but is in the range generating a warning
+    WARNING = 1
+    #: Value is known, valid, but is in the range generating an alarm condition
+    ALARM = 2
+    #: Value is known, but not valid, e.g. a RW before its first put
+    INVALID = 3
+    #: The value is unknown, for instance because the channel is disconnected
+    UNDEFINED = 4
+
+
+class ReadingMandatory(TypedDict):
+    value: Any
+    timestamp: float
+
+
+class Reading(ReadingMandatory, total=False):
+    severity: Severity
+    message: str
+
+
+class DescriptorMandatory(TypedDict):
+    source: str
+    dtype: Literal["string", "number", "array", "boolean", "integer"]
+    shape: List[int]
```

Comment:
What happens if the shape of an array changes mid-way through a scan?

Reply:
It shouldn't do. https://github.com/bluesky/event-model/pull/222 defines "bulk" events where the event is fixed size, but the number emitted is not, and there is a single pointer to a slice of an external file.